src/librustc/lint/builtin.rs

```diff
@@ -298,6 +298,12 @@ declare_lint! {
     "detects duplicate macro exports"
 }
 
+declare_lint! {
+    pub INTRA_LINK_RESOLUTION_FAILURE,
+    Warn,
+    "warn about documentation intra links resolution failure"
```

Comment:
As you prefer, I don't have a strong opinion on the naming.

Reply:
> intra links […] doesn't mean much on its own

What is this "intra links" thing you are talking about? Did you mean "intra _doc_ links" which has actual context for what the links are between/within and for which we also have an RFC? :trollface:

Reply:
@QuietMisdreavus Why "type links"? It also works for functions and macros etc.

Reply:
* I have never, since the RFC was accepted, referred to this feature with its full title in internal discussions. The shortened title has been used so much that the full one has fallen out of our collective memory, as evidenced in this lint title, this PR title, and my comment here.
* Calling it "type links" implies that it can link to "things that rustc cares about", even if the name is overly restrictive. Yes, i'm aware that the links can go to more than just "types".
* `INTRA_DOC_LINK_RESOLUTION_FAILURES` is *exceedingly* verbose, but a quick look at `rustc +nightly -W help` says it's in good company, so use that.